### PR TITLE
Prepare `v0.12.5`

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
I'll add the migrations later.

File: runtime/crab/src/migrations.rs [hunk 1]
```diff
@@ -23,6 +27,19 @@ impl OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
...
+	for precompile in CrabPrecompiles::<Runtime>::used_addresses() {
+		EVM::create_account(&precompile, vec![0x60, 0x00, 0x60, 0x00, 0xFD]);
+	}
+
+	let module = b"FromDarwiniaIssuing";
+	migration::remove_storage_prefix(module, b"MappingFactoryAddress", &[]);
+	migration::remove_storage_prefix(module, b"RemoteBackingAccount", &[]);
+
+	// TODO
+	// Do we need a migration for KtonTreasury?
```
Comment (AsceticBear) [inline runtime/crab/src/migrations.rs:41]: Any insight about this? @AurevoirXavier The other changes LGTM

Comment (AurevoirXavier) [reply runtime/crab/src/migrations.rs:41]: in progress

File: runtime/darwinia/src/migrations.rs [hunk 2]
```diff
@@ -23,90 +27,35 @@ impl OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
...
-				b"GameSamplePoints",
-			],
-		),
+	Scheduler::migrate_v2_to_v3();
+
+	for precompile in DarwiniaPrecompiles::<Runtime>::used_addresses() {
+		EVM::create_account(&precompile, vec![0x60, 0x00, 0x60, 0x00, 0xFD]);
+	}
+
+	let module = b"ToCrabBacking";
```
Comment (AsceticBear) [inline runtime/darwinia/src/migrations.rs:36]: I guess this is duplicated.

File: runtime/crab/src/migrations.rs [hunk 3]
```diff
@@ -23,6 +27,25 @@ impl OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
...
+		(b"FromDarwiniaIssuing", &[b"MappingFactoryAddress", b"RemoteBackingAccount"]),
+		(b"KtonTreasury", &[b"ProposalCount", b"Proposals", b"Approvals"]),
+	];
+	let hash = &[];
+
+	removed_items.iter().for_each(|(module, items)| {
+		items.iter().for_each(|item| migration::remove_storage_prefix(module, item, hash));
+	});
+
+	// TODO
```
Comment (AsceticBear) [inline runtime/crab/src/migrations.rs:46]: We can delete these comment now.

File: Cargo.toml [hunk 4]
```diff
@@ -107,3 +107,8 @@ panic = "unwind"
 [profile.release-lto]
 inherits = "release"
 lto      = true
+
+[patch.crates-io]
+evm           = { git = "https://github.com/darwinia-network/evm", branch = "darwinia-v0.12.4" }
```
Comment (AsceticBear) [inline Cargo.toml:112]: Sorry about this. The last one should be `branch = "darwinia-v0.12.5"` here.

**Review by AsceticBear — APPROVED** [review]

**Comment by AsceticBear** [comment]
@HackFisher Would you like to take a review?

File: runtime/darwinia/src/migrations.rs [hunk 5]
```diff
@@ -23,90 +27,29 @@ impl OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
...
-				b"KtonTokenAddress",
-				b"RedeemStatus",
-				b"LockAssetEvents",
+				b"SecureLimitedPeriod",
+				b"SecureLimitedRingAmount",
+				b"TransactionInfos",
+				b"RemoteMappingTokenFactoryAccount",
 			],
 		),
+		(b"KtonTreasury", &[b"ProposalCount", b"Proposals", b"Approvals"]),
```
Comment (HackFisher) [inline runtime/darwinia/src/migrations.rs:46]: Are there existing KTON balances in the KTONTreasury pallet account, do we need to burn them?

Comment (AurevoirXavier) [reply runtime/darwinia/src/migrations.rs:46]: Yes. But let's leave them there.

RING and KTON are using the same pallet account.

**Pull request merged**